lib/src/component_declaration/component_base.dart

```diff
@@ -91,6 +93,10 @@ ReactDartComponentFactoryProxy registerAbstractComponent(Type abstractComponentC
 /// via a fluent-style builder interface.
 typedef TProps UiFactory<TProps extends UiProps>([Map backingProps]);
 
+extension UiFactoryHelpers on UiFactory {
+  UiFactoryConfig<TProps> asForwardRefConfig<TProps extends bh.UiProps>({String displayName}) => UiFactoryConfig(propsFactory: PropsFactory.fromUiFactory(this), displayName: displayName);
+}
```

Comment:
The generic parameter can actually be specifiedd on the extension itself, making it so you don't have to duplicate the typing!
```suggestion
extension UiFactoryHelpers<TProps extends bh.UiProps> on UiFactory<TProps> {
  UiFactoryConfig<TProps> asForwardRefConfig({String displayName}) => UiFactoryConfig(propsFactory: PropsFactory.fromUiFactory(this), displayName: displayName);
}
```

Usage before:
```dart
Dom.button.asForwardRefConfig<DomProps>(displayName: 'FancyButton'),
```
Usage after
```dart
Dom.button.asForwardRefConfig(displayName: 'FancyButton'),
```

Reply:
Niice, much cleaner!

Reply:
Hmmm when I use that syntax, it doesn't pick up the underlying factory's generics value and has a return type of `UiFactoryConfig<UiProps>` when the generic should be a specific class (which yields this error: react-dom.development.js:3843 Uncaught Error: Type 'UiFactoryConfig<UiProps>' should be 'UiFactoryConfig<LogPropsProps>' to implement expected type 'UiFactoryConfig<LogPropsProps>'.). I poked around but didn't see a way to resolve that - does one come to your mind, @greglittlefield-wf?

Reply:
Hmm I'm seeing the return type work for me locally:
<img width="823" alt="Screen Shot 2020-08-05 at 3 10 06 PM" src="https://user-images.githubusercontent.com/1713967/89469180-c497bc00-d72d-11ea-955e-c65f24a1ad16.png">

I'm about to step away for a quick break, but I can try running an example with it after!

Reply:
Ahh, never mind @greglittlefield-wf. I found the issue and it works just fine! 🙂

Reply:
Noice!